spec/facter/resolvers/ec2_spec.rb

```diff
@@ -142,4 +142,19 @@
 
     ec2.resolve(:userdata)
   end
+
+  context 'when data received are ASCII-8BIT encoded' do
+    let(:headers) { { 'Accept' => '*/*' } }
+
+    it 'converts data received to UTF-8 encoding' do
+      allow(Facter::Resolvers::Ec2).to receive(:v2_token).and_return(nil)
+
+      bin_str = "nofail\xCC\xA6\"".strip.force_encoding(Encoding::ASCII_8BIT)
```

Comment:
You can also have a variable like `expected_str = "nofail\xCC\xA6\"".strip` which should be UTF-8, and then change the expectation to `expect(ec2.resolve(:userdata).encoding).to eql(expected_str)`. This way the string contents will also be checked.

Also, I think you can remove the `strip` calls since they don't change the string contents. It looks like the string is frozen so you can use `dup` instead for better readability.